packages/build/src/core/report_metrics.ts

```diff
@@ -0,0 +1,20 @@
+import { closeClient, formatTags, InputStatsDOptions, startClient, validateStatsDOptions } from '../report/statsd.js'
+
+/**
+ * Record number of functions build and differentiate between autogenerated and user generated.
+ * Sends to statsd daemon.
+ */
+export const reportMetrics = async function (statsdOpts: InputStatsDOptions, metrics): Promise<void> {
+  if (!validateStatsDOptions(statsdOpts)) {
+    return
+  }
+  console.log({ metrics })
+  if (metrics.length === 0) return
+  const client = await startClient(statsdOpts)
+
+  metrics.forEach((metric) => {
+    client[metric.type](metric.name, metric.value, formatTags(metric.tags))
```

Comment:
Can we add a guard around this in case a step returns a metric with an unsupported type, so that we don't get a system error?

```js
if (typeof client[metric.type] === "function") {
  // (...)
}
```

Reply:
That is a wonderful idea!